src/bundle/Controller/RecommendationController.php

```diff
@@ -60,6 +72,9 @@ public function showRecommendationsAction(Request $request): Response
         $response = new Response();
         $response->setPrivate();
 
+        $this->encoreTagRenderer->reset();
+        $this->entrypointLookupCollection->getEntrypointLookup('ezplatform')->reset();
+
```

Comment:
Assuming that this is done inside a sub-request of sorts, and for whatever reason the scripts/links are not treated separately by Symfony, you should also reset the internal cache **after** the response is generated. It is done so in the example provided:

```php
// src/Controller/SomeController.php

use Symfony\WebpackEncoreBundle\Asset\EntrypointLookupInterface;

class SomeController
{
    public function index(EntrypointLookupInterface $entrypointLookup)
    {
        $entrypointLookup->reset();
        // render a template

        $entrypointLookup->reset();  // <------ Note the second call
        // render another template

        // ...
    }
}
```

So what I think you want to do is store the response in a variable instead of returning it immediately, and call resets a second time. And only then return the response.

Without it, any scripts/links generated in the template will not be likewise present in the parent response. There is a practice where JS is included at the end of the HTML document, and that addresses it.